app/src/org/commcare/android/database/global/models/ApplicationRecord.java

```diff
@@ -56,4 +82,102 @@ public int getStatus() {
     public void setStatus(int status) {
         this.status = status;
     }
+
+    public String getUniqueId() {
+        return uniqueId;
+    }
+
+    public void setUniqueId(String id) {
+        this.uniqueId = id;
+    }
+
+    public String getDisplayName() {
+        return this.displayName;
+    }
+
+    public void setDisplayName(String appName) {
+        this.displayName = appName;
+    }
+
+    public int getVersionNumber() {
+        return this.versionNumber;
+    }
+
+    public void setVersionNumber(int version) {
+        this.versionNumber = version;
+    }
+
+    public void setArchiveStatus(boolean b) {
+        this.isArchived = b;
+    }
+
+    public boolean isArchived() {
+        return this.isArchived;
+    }
+
+    public void setResourcesStatus(boolean b) {
+        this.resourcesValidated = b;
+    }
+
+    public boolean resourcesValidated() {
+        return this.resourcesValidated;
+    }
+
+    // A 'visible' app record has status installed and is not archived
+    public boolean isVisible() {
+        return status == STATUS_INSTALLED && !isArchived;
+    }
+
+    // A 'usable' app record is one that will actually get shown to a user -- is visible and has

```

Comment:
Can you make these comments be in the /* style so they show up in javadocs/inline docs?